libcontainer/rootfs_linux.go

```diff
@@ -723,19 +730,23 @@ func checkProcMount(rootfs, dest, source string) error {
 	}
 	if path == "." {
 		// an empty source is pasted on restore
-		if source == "" {
+		if m == nil {
 			return nil
 		}
-		// only allow a mount on-top of proc if it's source is "proc"
-		isproc, err := isProc(source)
-		if err != nil {
-			return err
-		}
-		// pass if the mount is happening on top of /proc and the source of
-		// the mount is a proc filesystem
-		if isproc {
+		// only allow a mount on-top of proc if its source is a procfs mount ...
+		if m.Device == "proc" {
```

Comment:
I think maybe we should change to `isproc`, because in the code, the value of `m.Device` is always `bind` or other, but not `proc`.
https://github.com/opencontainers/runc/blob/22a7199bb7b9d1a92175b18c4db4fb46318f549b/libcontainer/rootfs_linux.go#L552-L553
https://github.com/opencontainers/runc/blob/22a7199bb7b9d1a92175b18c4db4fb46318f549b/libcontainer/rootfs_linux.go#L670-L671